app/lib/flag-route.js

```diff
@@ -1,16 +1,29 @@
-import Route from '@ember/routing/route';
+const ROUTE_LIX_ENABLED_KEY = '__route_lix_is_enabled';
 
 function generateRouteHandler(owner, key, treatmentRouteName) {
   const treatmentRoute = owner.lookup(`route:${treatmentRouteName}`);
   const flagService = owner.lookup('service:flags');
   return {
+    set(target, prop, value) {
+      const flagIsControl = !Reflect.get(target, ROUTE_LIX_ENABLED_KEY);
+      if (prop === 'controller') {
+        // Set the other controller too, because _internalReset assumes controller is set once route is setup
+        if (flagIsControl) {
+          Reflect.set(treatmentRoute, prop, owner.lookup(`controller:${treatmentRouteName}`));
+        } else {
+          Reflect.set(target, prop, owner.lookup(`controller:${Reflect.get(target, 'routeName')}`));
```

Comment:
Is this same to `Reflect.set(target, prop, value)` in this branch?

Reply:
No, because the `controller` is empty in some cases, when we need to make sure it's populated. This is because the underlying ember code assumes that once the route controller is set, it doesn't need to reset it.